cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,544 @@
+	if configmap.Data["allowunknown"] != "" {
+		allowUnknown, err := strconv.ParseBool(configmap.Data["allowunknown"])
+		if err == nil {
+			manager.allowUnknown = allowUnknown
+		} else {
+			glog.Errorf("Unable to parse allowunknown value: %s", allowUnknown)
+		}
+	}
+	if configmap.Data["reverify"] != "" {
+		reverify, err := strconv.Atoi(configmap.Data["reverify"])
+		if err == nil {
+			manager.recurring = reverify
+			// Trigger the reverification logic. If it's already in the
+			// middle of reverifying, drop the event - it'll handle it
+			// at the end of reverification.
+			select {
+			case manager.recurringChan <- reverify:
+			default:
+			}
+		} else {
+			glog.Errorf("Unable to parse reverify value: %s", reverify)
+		}
+	}
+}
+
+func run(stop <-chan struct{}) {
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: nodeAddFn,
+		},
+	)
+
+	_, configController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.ConfigMaps("kube-system").List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.ConfigMaps("kube-system").Watch(options)
+			},
+		},
+		&api.ConfigMap{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    configAddFn,
+			UpdateFunc: configUpdateFn,
+		},
+	)
+
+	_, policyController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.tpmhandler.PolicyClient.List(&options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.tpmhandler.PolicyClient.Watch(&options)
+			},
+		},
+		&runtime.Unstructured{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    policyAddFn,
+			UpdateFunc: policyUpdateFn,
+			DeleteFunc: policyDeleteFn,
+		},
+	)
+	err := loadPolicy()
+	if err != nil {
+		glog.Errorf("Unable to load PCR policy: %v", err)
+		os.Exit(1)
+	}
+	go reverify()
+	go nodeController.Run(wait.NeverStop)
+	go configController.Run(wait.NeverStop)
+	go policyController.Run(wait.NeverStop)
+	select {}
+}
+
+func main() {
+	var tpmhandler tpm.TPMHandler
+	var config restclient.Config
+
+	//	config, err := clientcmd.BuildConfigFromFlags(manager.Master, manager.Kubeconfig)
+	err := client.SetKubernetesDefaults(&config)
+	if err != nil {
+		fmt.Errorf("Unable to create client configuration: %v", err)
+		os.Exit(1)
+	}
+	config.Host = "http://localhost:8080"
```

Comment:
We may have spoke about this in person at some point, but why is this the only option? This restricts deployment options (must either be deployed in same pod as api-server, or must have host networking and be co-located with api-server which also needs hostNetworking). Not insurmountable -- but we lose flexibility here with this type of requirement.

Also, have we confirmed that "no user" actually means "insecure interface" and that is a safe assumption (I remember seeing this in the admission controller code at some point).

Alternatively is there some pattern here we might want to use where we give the tpm-controller special privileges which the admission controller respects?

/cc @ericchiang